packages/edit-navigation/src/components/header/inserter.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { Button, ToolbarItem } from '@wordpress/components';
+import { _x } from '@wordpress/i18n';
+import { useRef } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { plus } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { useNavigationEditorRootBlock } from '../../hooks';
+import { store as editNavigationStore } from '../../store';
+
+function Inserter() {
+	const inserterButton = useRef();
+	const { navBlockClientId } = useNavigationEditorRootBlock();
+
+	const { isInserterOpened, hasInserterItems } = useSelect( ( select ) => {
+		return {
+			hasInserterItems: select( blockEditorStore ).hasInserterItems(
+				navBlockClientId
+			),
+			isInserterOpened: select( editNavigationStore ).isInserterOpened(),
+		};
+	} );
+
+	const { setIsInserterOpened } = useDispatch( editNavigationStore );
+
+	return (
+		<ToolbarItem
+			ref={ inserterButton }
+			as={ Button }
+			className="edit-navigation-header-toolbar__inserter-toggle"
```

Comment:
So the CSS naming guidelines are that the classname should represent the package and component name, only sub-elements within the component should have an `__element`. 

That makes sense because the component should be reusable (at least within the header), so may not always be rendered inside the 'header-toolbar' part of the UI.

```suggestion
			className="edit-navigation-inserter"
```

That name could be too generic though, so an idea is to give the component a prefixed name based on the folder:
- `HeaderInserterToggle`
- `edit-navigation-header-inserter-toggle`

I feel like the toggle suffix is also worth it, as my mistake was to initially think this component is the inserter itself.

Reply:
I'll rename the component as well.

Reply:
There were issues with having a different css class for the inserter, with widget block editor and e2e tests. Just want to make sure we can make e2e utils like insert block ( that uses search to find the block btw ) will work here as well.

Reply:
I've updated the component to be `InserterToggle` and updated the CSS to `edit-navigation-header-inserter-toggle`.

Reply:
I'll check the e2e tests don't break and we can still locate the inserter. If we're relying on classnames to locate the inserter we need to rethink how we're writing our tests. If that's the case I'll rewrite to make the tests less dependent on developer APIs.